lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -1625,22 +1633,39 @@ def get_course_discussion_user_stats(
         order_by = order_by or UserOrdering.BY_ACTIVITY
         if order_by != UserOrdering.BY_ACTIVITY:
             raise ValidationError({"order_by": "Invalid value"})
-    course_stats_response = get_course_user_stats(course_key, {
+    params = {
         'sort_key': str(order_by),
         'page': page,
         'per_page': page_size,
-    })
+    }
+    comma_separated_usernames = matched_users_count = matched_users_pages = None
+    if username_search_string:
+        comma_separated_usernames, matched_users_count, matched_users_pages = get_usernames_from_search_string(
+            course_key, username_search_string, page, page_size
+        )
+        if not comma_separated_usernames:
+            return DiscussionAPIPagination(request, 0, 1).get_paginated_response({
+                "results": []
+            })
+        params['usernames'] = comma_separated_usernames
+
+    course_stats_response = get_course_user_stats(course_key, params)
     serializer = UserStatsSerializer(
         course_stats_response["user_stats"],
         context={"is_privileged": is_privileged},
         many=True,
     )
+    if comma_separated_usernames:
+        add_stats_for_users_with_no_discussion_content(
```

Comment:
What about omitting users that have not interacted with discussions rather than returning empty stats?

Reply:
We can't do that because the paginated data would be inconsistent that way. Say there are 5 matched users on a page that have no discussion content, the **cs_comment_service** will return 5 results only, and the data for the page (expected 10 but got 5) would be inconsistent.

Reply:
I think it would be cleaner if this method returns the updated data instead of mutating the passed parameter.